hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -459,6 +459,68 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Reconcile etcd client MTLS secret if the control plane is using an unmanaged etcd cluster
+	if hcluster.Spec.Etcd.ManagementType == hyperv1.Unmanaged {
+		r.Log.Info("Reconciling etcd client mtls secret")
+		reportFailureOnHostedClusterFunc := func(message string) {
```

Comment:
Sorry this isn't documented, but the design for this controller is such that all status in should be computed in discrete steps prior to this line:

https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L281

The HostedCluster Available condition specifically is computed here:

https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L203

Any changes to how HostedCluster availability is computed should happen there. I'm not sure that any of the changes in this PR should have an impact on the HostedCluster calculation, as the hypershift operator isn't actually managing etcd and instead uses the HostedControlPlane availability as a proxy for health at that layer right now.

So anyway, I think all the condition handling here should be removed, and if we want to incorporate etcd into status conditions, let's make that a followup discussion unless theres some specific need to incorporate it right now.

Hope that all makes sense.

Reply:
No worries @csrwng is this cool with you?

Reply:
yea this was a good catch thank you for this review